overthesun/simoc
# Separate `simoc-abm`

This PR includes changes to the backend for separating out the `simoc-abm` package. Included in this first commit:
- Add `simoc-abm` (test) as a requirement
- Update routes used by Configuration screen graphs: `get_energy` and `get_co2_o2`
- Update `convert_configuration` function
- Update main game initialization and data fetching loops in `views.py` and `tasks.py`

At this point, the backend appears to be doing everything properly on the configuration and dashboard screens, but the frontend needs an update to read the new schema of simoc-abm. Next steps will be:

1. Create a matching frontend PR in `simoc-web`
2. Create/test updated preset-simdata files
3. Create tests for the funcs mentioned above
4. Finalize `simoc_abm` and publish it on PyPl proper
5. Update this PR with #4, remove old agent model files, convert from Draft PR to real one.

At some point in the future, the stuff which is happening in `convert_configuration` should really happen in the frontend. But that's a discussion for another day.
